pkg/plugin/sdk/deployment.go

```diff
@@ -206,7 +206,33 @@ func (s *DeploymentPluginServiceServer[Config, DeployTargetConfig]) ExecuteStage
 		stageID:       request.GetInput().GetStage().GetId(),
 		logPersister:  lp,
 	}
-	return executeStage(ctx, s.base, &s.config, nil, client, request, s.logger) // TODO: pass the deployTargets
+
+	dtNames, err := request.GetInput().GetDeployment().GetDeployTargets(s.commonFields.config.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get deploy targets for plugin %s: %v", s.commonFields.config.Name, err)
+	}
```

Comment:
The plugin which doesn't have deploy target configurations such as WAIT will return an error in this if statement, and it's not expected behavior.

ref; https://github.com/pipe-cd/pipecd/blob/4afa3e61937502fd95614889cbd8294dd794fcd6/pkg/model/deployment.go#L217-L224

Reply:
@Warashi 
Thanks.
I fixed the method logic to allow the case of no deploy target. We can decide whether it is an error or not in the plugin logic from now.
[0e167c0](https://github.com/pipe-cd/pipecd/pull/5616/commits/0e167c0bc0bf7520bb7679e8c322cd6863c04373)